tests/cft/test_r2_support_bundle.py

```diff
@@ -756,6 +756,99 @@ def test_31256(self):
         self.LOGGER.info("Successfully validated Motr rotating log files are as per "
                          "frequency configured for all pods")
 
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.lc
+    @pytest.mark.support_bundle
+    @pytest.mark.tags("TEST-34548")
+    def test_34548(self):
+        """
+        Validate support bundle “size_limit” filter
+        """
+        self.LOGGER.info("STARTED: Test to validate support bundle 'size_limit' filter")
+
+        self.LOGGER.info("Step 1: Generate support bundle")
+
+        pod_list = self.node_obj.get_all_pods(pod_prefix=constants.POD_NAME_PREFIX)
+        machine_id = self.node_obj.get_machine_id_for_pod(pod_list[0])
+        output = self.node_obj.execute_cmd(cmd=comm.KUBECTL_GET_POD_CONTAINERS.format(pod_list[0]),
+                                           read_lines=True)
+        container_list = output[0].split()
+
+        self.LOGGER.info("Generating support bundle for pod: %s", pod_list[0])
+        sb_identifier = system_utils.random_string_generator(10)
+        self.LOGGER.info("Support Bundle identifier of : %s", sb_identifier)
+
+        resp = sb.gen_sb_with_param(sb_identifier, pod_list[0], container_list[0],
+                                    "TEST-34548", size_limit=f"{constants.SB_SIZE_MB}MB")
+        self.LOGGER.info("response of support bundle generation: %s", resp)
+        sb_local_path = os.path.join(os.getcwd(), "support_bundle_copy")
+
+        self.LOGGER.info("Step 2: Creating local directory")
+        if os.path.exists(sb_local_path):
+            self.LOGGER.info("Removing existing directory %s", sb_local_path)
+            shutil.rmtree(sb_local_path)
+        os.mkdir(sb_local_path)
+        self.LOGGER.info("sb copy path: %s", sb_local_path)
+
+        self.LOGGER.info("Step 3: Copy support bundle to local directory")
+        copy_sb_from_path = constants.R2_SUPPORT_BUNDLE_PATH + sb_identifier
+        sb_copy_path = "/root/support_bundle/"
+
+        copy_sb_cmd = comm.K8S_CP_TO_LOCAL_CMD.format(pod_list[0], copy_sb_from_path,
+                                                      sb_copy_path, container_list[0])
+        self.node_obj.execute_cmd(cmd=copy_sb_cmd, read_lines=True)
+
+        sb_copy_full_path = sb_copy_path + sb_identifier + "_" + machine_id + ".tar.gz"
+        sb_local_full_path = sb_local_path + "/" + sb_identifier + ".tar.gz"
+        self.node_obj.copy_file_to_local(sb_copy_full_path, sb_local_full_path)
+
+        self.LOGGER.info("Step 4: Get support bundle tar file size")
+        file_size_mb = os.path.getsize(sb_local_full_path) >> 20
+        self.LOGGER.info("file size: %s", file_size_mb)
+
+        if file_size_mb > constants.SB_SIZE_MB:
```

Comment:
what if its less than what is expected?

Reply:
Its ok to have less size, if that much logs are not there then it will be less.
Other case is also possible, there are binary files also which can not be reduced depending on size restriction.
Discussion is going on, what should be the minimum size it should support.
so will update it once that is finalized.